crypto/evp/m_sigver.c

```diff
@@ -61,41 +63,43 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
     if (e != NULL || locpctx->engine != NULL)
         goto legacy;
 
-    if (signature != NULL) {
-        if (!EVP_SIGNATURE_up_ref(signature))
-            goto err;
-    } else {
+    signature =
+        EVP_SIGNATURE_fetch(NULL, locpctx->algorithm, locpctx->propquery);
+    if (signature != NULL && locpctx->keymgmt == NULL) {
+        int name_id = EVP_SIGNATURE_number(signature);
+
+        keymgmt =
+            evp_keymgmt_fetch_by_number(NULL, name_id, locpctx->propquery);
+    }
+
+    if (keymgmt == NULL
```

Comment:
Can it be the case that `locpctx->keymgmt != NULL` but `keymgmt == NULL`? In which case do we need to check `locpctx->keymgmt` here?

Reply:
Hmmm... You're right, something is off. Not sure that your suggestion is right, though. Gimme a moment to think it over.